### PR TITLE
Some build "improvements"

### PR DESCRIPTION
**Pull request by bennybp**
1. Allows for building of shared or static libraries, and for static libraries with PIC enabled
2. Puts TBB's flags and libs directly in the config file, removing the need to distribute the CMake modules
3. Updates Cereal and downloads it if needed, and uses cereal's CMake config file
4. Installs libtaskforce into the standard directory hierarchy. Flattens cereal's install into the same hierarchy (rather than into an external dir)
5. Puts TBB's flags and libs directly in the config file. Combined with (3), this removes the need to distribute the CMake modules

The first commit (1f2cddb) just removes cereal from the source tree. You are probably mostly interested in the second (e667c0d)

### CODE REVIEW

**Comment by bennybp** [comment]
I think this is ready for any comments from you.

It's interesting that Lori and I had the same idea - to remove "external" install directories

File: CMakeLists.txt [hunk 1]
```diff
@@ -25,6 +25,16 @@ cmake_minimum_required(VERSION 3.1.3)
 project(LibTaskForce LANGUAGES CXX)
 set(CMAKE_CXX_EXTENSIONS False CACHE BOOL "Enable/Disable compiler-specific C++ extensions")
 set(CMAKE_CXX_STANDARD 14 CACHE STRING "Which C++ standard to use")
+set(LIBTASKFORCE_BUILD_SHARED FALSE CACHE BOOL "Build a shared library, rather than a static library")
```
Comment (ryanmrichard) [inline CMakeLists.txt]: Is there a reason this doesn't use ` BUILD_SHARED_LIBS`

File: CMakeLists.txt [hunk 2]
```diff
@@ -25,6 +25,16 @@ cmake_minimum_required(VERSION 3.1.3)
...
+set(LIBTASKFORCE_BUILD_PIC FALSE CACHE BOOL "Build the static libraric with PIC enabled")
```
Comment (ryanmrichard) [inline CMakeLists.txt]: ...and this doesn't use `CMAKE_POSITION_INDEPENDENT_CODE`

**Comment by loriab** [comment]
actually, I've been considering it for awhile. then i saw you do it. and i needed it for simple pybind11 detection for psi4 plugins. so it happened. I've been slightly monitoring your build changes. My psi4 changes are pretty much wrapped up for now. (except mathlibs)

File: CMakeLists.txt (hunk 1 above)
Comment (bennybp) [inline CMakeLists.txt]: There's a very good reason for that - I didn't know about that variable. That might clean up bphash/bpprint also

File: CMakeLists.txt (hunk 2 above)
Comment (bennybp) [inline CMakeLists.txt]: This one I did know about. I suppose it's nitpicky, but I wanted to set PIC only for the library and not the test executables. CMAKE_POSITION_INDEPENDENT_CODE sets it for all libraries and executables. But it doesn't harm anything if you'd like me to change it back

File: CMakeLists.txt (hunk 1 above)
Comment (bennybp) [inline CMakeLists.txt]: Fixed in recent push

**Comment by ryanmrichard** [comment]
I learned while rewriting Psi4's build that CMake has a variable for almost everything you can think of when it comes to build options.  The one thing that I can't find concretely is whether it already has built in support for install staging.  There's a couple of places where I find hints that it might, which of course I'm hard pressed to find those hints now...

**Comment by ryanmrichard** [comment]
Is this ready to be merged then?

**Comment by bennybp** [comment]
They should be (if you approve of the other changes).

**Pull request merged**